library/std/src/os/unix/process.rs

```diff
@@ -12,6 +12,16 @@ use crate::sealed::Sealed;
 use crate::sys;
 use crate::sys_common::{AsInner, AsInnerMut, FromInner, IntoInner};
 
+#[cfg(not(any(target_os = "vxworks", target_os = "espidf", target_os = "horizon")))]
+type UserId = u32;
+#[cfg(not(any(target_os = "vxworks", target_os = "espidf", target_os = "horizon")))]
+type GroupId = u32;
+
+#[cfg(any(target_os = "vxworks", target_os = "espidf", target_os = "horizon"))]
+type UserId = u16;
```

Comment:
This changes the signature for `target_os = "vxworks"`, doesn't it? I don't think that's allowed.

Why is it necessary to change the signature depending on the target OS? Can't we keep the IDs u32, and extending it / truncating (with checks perhaps) it internally?

Reply:
Oops, I misread the old code. It's already a conditional typedef. Ignore my comment.